core/src/main/java/org/apache/iceberg/BaseRewriteManifests.java

```diff
@@ -295,7 +317,8 @@ private WriterWrapper getWriter(Object key) {
   @Override
   protected void cleanUncommitted(Set<ManifestFile> committed) {
     cleanUncommitted(newManifests, committed);
-    cleanUncommitted(addedManifests, committed);
+    // rewritten manifests are always owned by the table
+    cleanUncommitted(rewrittenAddedManifests, committed);
```

Comment:
Nit: it would be nice to explain why `addedManifests` is not cleaned up: because they are added to the manifest list and are not compacted.